storage/storage_dest.go

```diff
@@ -416,23 +432,37 @@ func (s *storageImageDestination) tryReusingBlobAsPending(digest digest.Digest,
 		}
 	}
 
-	tocDigest := digest
-	if options.TOCDigest != nil {
-		tocDigest = *options.TOCDigest
-	}
+	if options.TOCDigest != "" && options.LayerIndex != nil {
+		// we are already holding s.lock
+		diffOutput, ok := s.diffOutputs[blobDigest]
+		if ok {
+			// Save this for completeness.
+			s.indexToTocDigest[*options.LayerIndex] = options.TOCDigest
+			return true, private.ReusedBlob{
+				Digest:    blobDigest,
+				TOCDigest: diffOutput.TOCDigest,
+				Size:      diffOutput.Size,
```

Comment:
(If this case remains, which I’m unsure about:) `Size` needs to be consistent with `blobDigest`, i.e. `size`.

Reply:
#2288 removed this case.